packages/kbn-securitysolution-utils/src/path_validations/index.ts

```diff
@@ -52,39 +52,60 @@ export enum OperatingSystem {
 export type EntryTypes = 'match' | 'wildcard' | 'match_any';
 export type TrustedAppEntryTypes = Extract<EntryTypes, 'match' | 'wildcard'>;
 
-export const validateFilePathInput = ({
+export const validateWildcardInput = ({
+  field = '',
   os,
   value = '',
 }: {
+  field?: string;
   os: OperatingSystem;
   value?: string;
 }): string | undefined => {
   const textInput = value.trim();
+  if (field === 'file.path.text') {
+    return validateFilePathInput({ os, value: textInput });
+  }
+  return validatePotentialWildcardInput(textInput);
+};
+
+const validateFilePathInput = ({
+  os,
+  value,
+}: {
+  os: OperatingSystem;
+  value: string;
+}): string | undefined => {
   const isValidFilePath = isPathValid({
     os,
     field: 'file.path.text',
     type: 'wildcard',
-    value: textInput,
+    value,
   });
   const hasSimpleFileName = hasSimpleExecutableName({
     os,
     type: 'wildcard',
-    value: textInput,
+    value,
   });
 
-  if (!textInput.length) {
+  if (!value.length) {
     return FILEPATH_WARNING;
   }
 
   if (isValidFilePath) {
     if (hasSimpleFileName !== undefined && !hasSimpleFileName) {
-      return FILENAME_WILDCARD_WARNING;
+      return WILDCARD_WARNING;
```

Comment:
I think I would expect to see only FILEPATH_WARNING in this function as it only deals with `file.path.text` field. What would make this simpler is if we combine the two if blocks in this function as

```json5
if (
    !isValidFilePath ||
    !value.length ||
    (hasSimpleFileName !== undefined && !hasSimpleFileName)
  ) {
    return FILEPATH_WARNING;
  }
```
and that's all the function should return.

Reply:
The `FILEPATH_WARNING` warns about a malformed path whereas the `WILDCARD_WARNING` warns the user that a wildcard will affect performance. We actually still want the `WILDCARD_WARNING` in this case, given that the path is valid.

Reply:
Oh yeah you're absolutely right! Perhaps the following may be simpler to reason with:

```json5
 if (!isValidFilePath || !value.length) {
   return FILEPATH_WARNING;
 }

 if (hasSimpleFileName !== undefined && !hasSimpleFileName) {
   return WILDCARD_WARNING;
 }
```